src/lib/services/chat/models/llamaindex/openai.py

```diff
@@ -88,17 +105,73 @@ def invoke(self, message: str) -> 'LlamaIndexOpenAIModel.Result':
             logger.error(self.result.error_message)
         return self.result
 
-    def get_model(self) -> 'LlamaIndexOpenAIModel.Result':
-        """
-        Return the LLM model instance.
+    def stream(self, messages: Any) -> Iterator[str]:
+        '''
+        Synchronously stream the model response token by token.
 
-        :return: Result object containing the model instance.
-        """
-        self.result.model = self.model
-        if self.model:
+        :param messages: Message string or list formatted for the model.
+        :return: Iterator yielding response chunks.
+        '''
+        try:
+            normalized_messages = self._normalize_messages(messages)
+            for chunk in self.model.stream_chat(normalized_messages):
+                yield chunk.delta
+        except Exception as e:  # pylint: disable=W0718
+            logger.error(f"Streaming error: {e}")
+            raise
+
+    async def ainvoke(self, messages: Any) -> 'LlamaIndexOpenAIModel.Result':
+        '''
+        Asynchronously invoke the model with a list of messages.
+
+        :param messages: Message string or list formatted for the model.
+        :return: Result object with content and metadata.
+        '''
+        try:
             self.result.status = "success"
-            logger.debug(f"Returned model '{self.config.model_name}'")
-        else:
+            normalized_messages = self._normalize_messages(messages)
+            response = await self.model.achat(normalized_messages)
+            self.result.content = response.text
+            self.result.metadata = response.additional_kwargs
+            logger.debug(f"Async prompt generated: {self.result.content}")
+        except Exception as e:  # pylint: disable=W0718
             self.result.status = "failure"
-            logger.error("No model present")
+            self.result.error_message = f"Async error: {e}"
+            logger.error(self.result.error_message)
         return self.result
+
+    async def astream(self, messages: Any) -> AsyncIterator[str]:
+        '''
+        Asynchronously stream the model response token by token.
+
+        :param messages: Message string or list formatted for the model.
+        :return: Async iterator yielding response chunks.
+        '''
+        try:
+            normalized_messages = self._normalize_messages(messages)
+            # ✅ Fix: await the coroutine that returns an async generator
```

Comment:
Depending on some peoples editors emojis may not render correctly within code comments. I'm fine with leaving it just in past experience with working with many devs that has been a problem. I think they're good to have in markdown files though